lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/LibertyUtils.java

```diff
@@ -253,19 +253,24 @@ public static String getVersion(String serverXMLUri) {
      * Return temp directory to store generated feature lists and schema. Creates
      * temp directory if it does not exist.
      * 
-     * @param folder WorkspaceFolderURI indicates where to create the temporary
-     *               directory
+     * @param libertyWorkspace LibertyWorkspace where the temporary directory is made
+     *               
      * @return temporary directory File object
      */
-    public static File getTempDir(String workspaceFolderURI) {
-        if (workspaceFolderURI == null) {
+    public static File getTempDir(LibertyWorkspace libertyWorkspace) {
+        if (libertyWorkspace == null) {
             return null;
         }
         try {
-            URI rootURI = new URI(workspaceFolderURI);
-            Path rootPath = Paths.get(rootURI);
-            File file = rootPath.toFile();
-            File libertyLSFolder = new File(file, ".libertyls");
+            Path targetFolderPath = findFileInWorkspace(libertyWorkspace, Paths.get("target"));
+            Path buildFolderPath  = findFileInWorkspace(libertyWorkspace, Paths.get("build"));
+
+            File libertyLSFolder = null;
+            if (targetFolderPath!= null && targetFolderPath.toFile().exists()) {
+                libertyLSFolder = new File(targetFolderPath.toFile(), ".libertyls");
+            } else if (buildFolderPath!= null && buildFolderPath.toFile().exists()) {
+                libertyLSFolder = new File(buildFolderPath.toFile(), ".libertyls");
+            }
 
             if (!libertyLSFolder.exists()) {
```

Comment:
Yeah, but I think we would return null in that case anyway. I'll add a check. Shouldn't change the outcome there.

Reply:
Returning null would cause the calling methods to fail. Previously it could not return null. Perhaps we create in the old location if target/build are not found?

Reply:
Actually, it did return null before but only when the workspace was null.